### PR TITLE
Add formatters and linters to the CI and fix the fallout

### PR DESCRIPTION
**Pull request by hnez**
This PR adds black, isort and prettier (auto formatters) and flake8 (python linter), as well as codespell to check for typos and fixes all issues that the various tools brought up.

TODO before merging:

  - [x] ~~This PR is now based on #34, which already added black and flake8. That PR should be merged first and then this one should be rebased on top of it.~~
        This PR is no longer based on #34, because it now uses `ruff` instead of `black` + `flake8` + `isort`, which has the benefit of having one tool and not three tools fighting about the question what python code should look like.

### CODE REVIEW

**Comment by hnez** [comment]
I've only noticed that this duplicates work that was already done in #34.
I will rebase my work on top of it, make sure that everything works and if so add my approval to that PR so that we can get it merged, because it also addressed a few things that I did not spot.

**Comment by SmithChart** [comment]
> I've only noticed that this duplicates work that was already done in #34. I will rebase my work on top of it, make sure that everything works and if so add my approval to that PR so that we can get it merged, because it also addressed a few things that I did not spot.

Feel free to pick the changes you missed from there and add them to your PR. I don't mind if we drop my PR in favor of yours.

File: pyproject.toml [hunk 1]
```diff
@@ -5,3 +5,8 @@ line-length = 119
 max-line-length = 119
 count = true
 exclude = "build,debian,dist,envs,fastentrypoints.py,__pycache__,usbmuxctl.egg-info,venv,contrib,setup.py,env-qa,env,env-packaging"
+
+[tool.isort]
+line_length = 119
+multi_line_output = 3
+include_trailing_comma = true
```
Comment (SmithChart) [inline pyproject.toml:12]: Please also add the tool to the `qa`-target in the `Makefile`.

File: .github/workflows/prettier.yaml
Comment (SmithChart) [inline .github/workflows/prettier.yaml:1]: Please split the adding of the tool and changing of the files into two separate commits.
This way one can ignore the commit with the format changes when e.g. `git blame` a file. But will still see the changes tot he tooling.

File: .github/workflows/prettier.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: latest
+      - run: npm i -D prettier prettier-plugin-toml
+      - run: npx prettier --check pyproject.toml
+      - run: npx prettier --check lxa_iobus/server/static/index.html
+      - run: npx prettier --check lxa_iobus/server/static/main.js
+      - run: npx prettier --check lxa_iobus/server/static/style.css
```
Comment (SmithChart) [inline .github/workflows/prettier.yaml:18]: Please also add this to the `qa`-target in the `Makefile`.

File: .github/workflows/codespell.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+on: [push, pull_request]
+
+jobs:
+  codespell:
+    name: Codespell
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - run: python3 -m pip install codespell
+      - run: codespell
```
Comment (SmithChart) [inline .github/workflows/codespell.yaml:12]: I like the idea of running codespell.

Please split the introduction of the tooling and the changes needed to make it happy into two commits.

Comment (SmithChart) [inline .github/workflows/codespell.yaml:12]: Please also add to the `qa`-target in the `Makefile`.

File: lxa_iobus/canopen.py [hunk 4]
```diff
@@ -94,11 +94,6 @@ def __init__(self, **kwargs):
             setattr(self, key, value)
 
 
-# lss_adresse: sowas wie mac adresse (wie eingebrannt)
-# id_number  : (unbekannt, kann man wahrscheinlich nicht verstehen)
-# node_id    : temporaere adresse auf dem bus (wie ip unter dhcp)
-
-
```
Comment (SmithChart) [inline lxa_iobus/canopen.py:103]: Unrelated change.

**Review by SmithChart — CHANGES_REQUESTED** [review]
Please see change-requests.

Maybe we also want to add `codespell` and `isort`  to the other python-based tools?

File: pyproject.toml (hunk 1 above)
Comment (hnez) [reply pyproject.toml]: Has been replaced by `ruff`, which does linting, formatting and import sorting and is now part of the `qa` target in the Makefile.

File: .github/workflows/codespell.yaml (hunk 3 above)
Comment (hnez) [reply .github/workflows/codespell.yaml]: Yeah, i feared that it would have many false positives, but it actually had none.
I've now split the spell checking and tooling/CI changes into two commits.

Comment (hnez) [reply .github/workflows/codespell.yaml]: Done. `make qa-codespell` just runs codespell, `make qa` runs all checks, including codespell.

File: lxa_iobus/canopen.py (hunk 4 above)
Comment (hnez) [reply lxa_iobus/canopen.py:103]: Split into its own commit.

File: .gitignore [hunk 5]
```diff
@@ -225,12 +225,14 @@ celerybeat.pid
 # Environments
 .env
 .venv
+.
```
Comment (SmithChart) [inline .gitignore:228]: This looks odd.

**Review by SmithChart — APPROVED** [review]

File: .gitignore (hunk 5 above)
Comment (hnez) [reply .gitignore]: Yes, yes it does. Should be fixed.

**Review by SmithChart — APPROVED** [review]
